### PR TITLE
Add path system to reference nested data in process

### PR DESCRIPTION
**Pull request by krhubert**
+ small refactor

dep https://github.com/mesg-foundation/engine/pull/1540
close https://github.com/mesg-foundation/engine/issues/1306

### CODE REVIEW

File: orchestrator/orchestrator.go [hunk 1]
```diff
@@ -223,3 +260,42 @@ func (s *Orchestrator) processTask(nodeKey string, task *process.Process_Node_Ta
 	}, s.accountName, s.accountPassword)
 	return err
 }
+
+func query(data *types.Struct, path *process.Process_Node_Map_Output_Reference_Path) (*types.Value, error) {
```
Comment (NicolasMahe) [inline orchestrator/orchestrator.go]: Let's rename query:
```suggestion
func resolveRef(data *types.Struct, path *process.Process_Node_Map_Output_Reference_Path) (*types.Value, error) {
```

File: orchestrator/orchestrator.go [hunk 2]
```diff
@@ -223,3 +260,42 @@ func (s *Orchestrator) processTask(nodeKey string, task *process.Process_Node_Ta
...
+
+	v, ok = data.Fields[key.Key]
+	if !ok {
+		return nil, fmt.Errorf("orchestrator: key %s not found", key.Key)
+	}
+
+	for p := path.Path; p.Path != nil; p = p.Path {
+		switch s := p.Selector.(type) {
+		case *process.Process_Node_Map_Output_Reference_Path_Key:
+			str, ok := v.GetKind().(*types.Value_StructValue)
```
Comment (NicolasMahe) [inline orchestrator/orchestrator.go]: You should use the helper function generated by proto. something like:
```suggestion
			str, ok := v.GetStructValue()
```

File: orchestrator/orchestrator.go [hunk 3]
```diff
@@ -223,3 +260,42 @@ func (s *Orchestrator) processTask(nodeKey string, task *process.Process_Node_Ta
...
+	}
+
+	v, ok = data.Fields[key.Key]
+	if !ok {
+		return nil, fmt.Errorf("orchestrator: key %s not found", key.Key)
+	}
+
+	for p := path.Path; p.Path != nil; p = p.Path {
+		switch s := p.Selector.(type) {
+		case *process.Process_Node_Map_Output_Reference_Path_Key:
```
Comment (NicolasMahe) [inline orchestrator/orchestrator.go]: you could transform this case to a simple if / else if / else.
At least, add a `default` that return an error.

**Comment by NicolasMahe** [comment]
optional path test added

**Comment by NicolasMahe** [comment]
TODO:
- [ ] add e2e test on nested ref. the only e2e test is testing only 1 level of nested ref.

File: orchestrator/orchestrator.go [hunk 4]
```diff
@@ -224,9 +225,9 @@ func (s *Orchestrator) resolveInput(wfHash hash.Hash, exec *execution.Execution,
 		if err != nil {
 			return nil, err
 		}
-		return s.resolveInput(wfHash, parent, nodeKey, outputKey)
+		return s.resolveInput(wfHash, parent, nodeKey, path)
```
Comment (antho1404) [inline orchestrator/orchestrator.go]: `resolveInput` wasn't resolving any path, I just added a commit with the fix

**Pull request merged**